runtime/doc/dev_vimpatch.txt

```diff
@@ -308,6 +308,9 @@ FILETYPE DETECTION                 *dev-vimpatch-filetype*
 Nvim's filetype detection behavior matches Vim, but is implemented as part of
 |vim.filetype| (see $VIMRUNTIME/lua/vim/filetype.lua).
 
+Prefer explicit filenames/extensions over patterns, especially for case
+insensitive matches.
```

Comment:
Maybe expand a little and give examples? This should be enough guidance for new contributors even without any of us around.

Reply:
(And if you link here, the code comment can be simplified.)

Reply:
Force pushed some examples while also linking to this PR.